tests/src/test/java/org/sonarqube/tests/user/SonarCloudNotificationsWsTest.java

```diff
@@ -0,0 +1,110 @@
+
+    assertThat(list.getNotificationsList()).isEmpty();
+    assertThat(list.getChannelsList()).containsExactlyInAnyOrder("EmailNotificationChannel");
+    assertThat(list.getGlobalTypesList())
+      .containsExactlyInAnyOrder(
+        "CeReportTaskFailure", "ChangesOnMyIssue", "SQ-MyNewIssues")
+      .doesNotContain("NewAlerts", "NewFalsePositiveIssue", "NewIssues");
+    assertThat(list.getPerProjectTypesList()).containsExactlyInAnyOrder(
+      "CeReportTaskFailure", "ChangesOnMyIssue", "NewAlerts", "NewFalsePositiveIssue", "NewIssues", "SQ-MyNewIssues");
+  }
+
+  @Test
+  public void add_global_and_project_notifications() {
+    Users.CreateWsResponse.User user = tester.users().generate();
+    assertThat(tester.as(user.getLogin()).wsClient().notifications().list(new ListRequest()).getNotificationsList()).isEmpty();
+    Projects.CreateWsResponse.Project project = tester.projects().provision();
+
+    tester.as(user.getLogin()).wsClient().notifications().add(new AddRequest().setChannel("EmailNotificationChannel").setType("ChangesOnMyIssue"));
+    tester.as(user.getLogin()).wsClient().notifications().add(new AddRequest().setChannel("EmailNotificationChannel").setType("NewIssues").setProject(project.getKey()));
+
+    assertThat(tester.as(user.getLogin()).wsClient().notifications().list(new ListRequest()).getNotificationsList())
+      .extracting(Notification::getChannel, Notification::getType, Notification::getProject, Notification::getProjectName)
+      .containsExactlyInAnyOrder(
+        tuple("EmailNotificationChannel", "ChangesOnMyIssue", "", ""),
+        tuple("EmailNotificationChannel", "NewIssues", project.getKey(), project.getName()));
+  }
+
+  @Test
+  public void remove_global_and_project_notifications() {
+    Users.CreateWsResponse.User user = tester.users().generate();
+    assertThat(tester.as(user.getLogin()).wsClient().notifications().list(new ListRequest()).getNotificationsList()).isEmpty();
+    Projects.CreateWsResponse.Project project = tester.projects().provision();
+    tester.as(user.getLogin()).wsClient().notifications().add(new AddRequest().setChannel("EmailNotificationChannel").setType("ChangesOnMyIssue"));
+    tester.as(user.getLogin()).wsClient().notifications().add(new AddRequest().setChannel("EmailNotificationChannel").setType("NewIssues").setProject(project.getKey()));
+
+    tester.as(user.getLogin()).wsClient().notifications().remove(new RemoveRequest().setChannel("EmailNotificationChannel").setType("ChangesOnMyIssue"));
+    tester.as(user.getLogin()).wsClient().notifications().remove(new RemoveRequest().setChannel("EmailNotificationChannel").setType("NewIssues").setProject(project.getKey()));
+
+    assertThat(tester.as(user.getLogin()).wsClient().notifications().list(new ListRequest()).getNotificationsList()).isEmpty();
+  }
+
+  @Test
+  public void fail_to_add_global_new_issues_notification() {
```

Comment:
remark: I'd test it at unit level and not at IT level

Reply:
yes, and no... It's minor, but it's not costly so why not ?